basic/airdrop-with-withdraw/src/lib.rs

```diff
@@ -0,0 +1,91 @@
+use scrypto::prelude::*;
+
+#[derive(NftData)]
+pub struct AirdropWithWithdrawData {
+    amount: Decimal,
+    token_type : Address, 
+    #[scrypto(mutable)]
+    is_collected : bool
+}
+
+blueprint! {
+
+    
+    struct AirdropWithWithdraw {
+        admin_badge: Address,
+        tokens : Vault,
+        recipient_address_by_badge_id : HashMap<u128, Address>,
```

Comment:
This does not have to be a HashMap, you are only using it to check if a NFT id is present in the keys. You could use a `HashSet<u128>` instead

Reply:
I removed the hasmap  because checking of  nft_data.is_collected is enough